File: programmers/단속 카메라.js
function solution(routes) {
  let answer = 0;

  routes.sort((a, b) => a[1] - b[1]);

  let cur = -30000;

  for (let i = 0; i < routes.length; i++) {
    let [s, e] = routes[i];

    if ((cur >= s) & (cur <= e)) {
      continue;
    } else {
      answer += 1;
      cur = e;
    }
  }

  return answer;
}

File: programmers/단속 카메라.py
def solution(routes):
    answer = 0
    mapp = []
    routes.sort(key=lambda x: (x[1], x[0]))
    
    curEnd = -1e10
    
    for r in routes:
        s, e = r
        
        if s>curEnd:
            answer+=1
            curEnd = e
        
    
    return answer